extensions/blocks/opentable/edit.js

```diff
@@ -57,6 +58,28 @@ function OpenTableEdit( {
 
 	const { align, rid, style, iframe, domain, lang, newtab } = attributes;
 
+	// We only want to show the warning about wide display when alignment changes, not on every page load
+	// so below ref is set to indicate the initial component mount
+	const mount = useRef( true );
+	useEffect( () => {
+		if ( mount.current ) {
+			mount.current = false;
+			return;
+		}
+		noticeOperations.removeAllNotices();
+		if ( style === 'wide' && align && align !== 'wide' ) {
```

Comment:
This needs a few more checks!
- The notice shouldn't appear on the wide style preview in the sidebar.
- The notice shouldn't appear on the placeholder, where we already use `noticeUI` for embed errors.

I've had success with this:

```js
const { noticeOperations } = props;
const { __isBlockPreview, align, rid, style } = attributes;

useEffect( () => {
	noticeOperations.removeAllNotices();
	if ( ! isEmpty( rid ) && ! __isBlockPreview && 'wide' === style && 'wide' !== align ) {
		noticeOperations.createNotice( {
			status: 'warning',
			content: __( 'Warning' ),
		} );
	}
}, [ __isBlockPreview, align, noticeOperations, rid, style ] );
```

Notes:
- `isEmpty( rid )` is what we use for determining if the block is in placeholder state. Since we use it a few times, we might as well assign it to a more clear variable, e.g. `const isPlaceholder = empty( rid )`.
- `__isBlockPreview` is part of our own `BlockStylesSelector` component, and turns out to be super handy in this case!
- We need to pass the actual used variables to the hook, not just the `attributes` object, or we would trigger this every time any attribute changes.

Reply:
Ahh, thanks for all this extra detail ... I still have a lot to learn about the jetpack blocks, been stuck in the back end for too long! I have made all these changes and seems much more robust now.